middleware/http/opa/middleware_test.go

```diff
@@ -429,3 +430,85 @@ func TestStatus_UnmarshalJSON(t *testing.T) {
 		})
 	}
 }
+
+func TestHandleRegoResult(t *testing.T) {
+	m := &Middleware{}
+	meta := &middlewareMetadata{
+		DefaultStatus: http.StatusForbidden,
+	}
+
+	tests := []struct {
+		name           string
+		result         any
+		expectedStatus int
+		expectedAllow  bool
+	}{
+		{
+			name:           "bool_result_allow",
+			result:         true,
+			expectedStatus: 0,
+			expectedAllow:  true,
+		},
+		{
+			name:           "bool_result_deny",
+			result:         false,
+			expectedStatus: http.StatusForbidden,
+			expectedAllow:  false,
+		},
+		{
+			name: "rego_result_allow",
+			result: map[string]any{
+				"allow":              true,
+				"additional_headers": map[string]string{"X-Test-Header": "test-value"},
+			},
+			expectedStatus: http.StatusOK,
+			expectedAllow:  true,
+		},
+		{
+			name: "rego_result_deny",
+			result: map[string]any{
+				"allow":              false,
+				"status_code":        301,
+				"additional_headers": map[string]string{"location": "test-value"},
+			},
+			expectedStatus: 301,
+			expectedAllow:  false,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			w := httptest.NewRecorder()
+			req := httptest.NewRequest(http.MethodGet, "/", nil)
+
+			allow := m.handleRegoResult(w, req, meta, tc.result)
+
+			assert.Equal(t, tc.expectedAllow, allow)
+			resp := w.Result()
+			defer func(Body io.ReadCloser) {
+				err := Body.Close()
+				if err != nil {
+					t.Fatal(err)
+				}
+			}(resp.Body)
```

Comment:
```suggestion
			defer resp.Body.Close()
```
This would fix the lint error